aiohttp/connector.py

```diff
@@ -1117,6 +1117,13 @@ def drop_exception(fut: "asyncio.Future[List[Dict[str, Any]]]") -> None:
             port = hinfo["port"]
 
             try:
+                server_hostname = None
+                if sslcontext:
+                    server_hostname = hinfo["hostname"]
+                    if server_hostname.endswith("."):
+                        # Strip trailing dot, certificates contain FQDN without dot.
+                        # See https://github.com/aio-libs/aiohttp/issues/3636
+                        server_hostname = server_hostname[:-1]
```

Comment:
Let's improve readability by getting rid of two unnecessary nesting levels:
```suggestion
                # Strip trailing dot, certificates contain FQDN without dot.
                # See https://github.com/aio-libs/aiohttp/issues/3636
                server_hostname = hinfo["hostname"].rstrip(".") if sslcontext else None
```